File: Backend/apps/notifications/api/serializers/notification.py
from rest_framework import serializers

from apps.notifications.models import Notification, NotificationPreference


class NotificationSerializer(serializers.ModelSerializer):
    class Meta:
        model = Notification
        fields = ["id", "notification_type", "title", "body", "is_read", "read_at", "related_order", "created_at"]
        read_only_fields = fields


class NotificationPreferenceSerializer(serializers.ModelSerializer):
    class Meta:
        model = NotificationPreference
        fields = ["sms_enabled", "email_enabled", "push_enabled", "whatsapp_enabled", "promo_opt_in"]

File: frontend/types/notification.ts
export type NotificationType = "ORDER_UPDATE" | "PAYMENT" | "PROMO" | "CHAT" | "SYSTEM";

/** Mirrors apps.notifications.api.serializers.notification.NotificationSerializer */
export interface Notification {
  id: number;
  notification_type: NotificationType;
  title: string;
  body: string;
  is_read: boolean;
  read_at: string | null;
  related_order: number | null;
  created_at: string;
}

/** Mirrors apps.notifications.models.NotificationPreference + its serializer */
export interface NotificationPreference {
  sms_enabled: boolean;
  email_enabled: boolean;
  push_enabled: boolean;
  whatsapp_enabled: boolean;
  promo_opt_in: boolean;
}

export type UpdateNotificationPreferenceInput = Partial<NotificationPreference>;

export type DevicePlatform = "IOS" | "ANDROID" | "WEB";

/** Mirrors apps.notifications.api.serializers.device_token.DeviceTokenSerializer */
export interface DeviceToken {
  id: number;
  token: string;
  platform: DevicePlatform;
  is_active: boolean;
  created_at: string;
}

/** Payload for POST /notifications/device-tokens/ */
export interface RegisterDeviceTokenInput {
  token: string;
  platform: DevicePlatform;
}
